client/src/components/WorkflowInvocationState/WorkflowInvocationOverview.vue

```diff
@@ -108,8 +108,10 @@ const invocationPdfLink = computed<string | null>(() => {
     }
 });
 
-const hasMessages = computed<boolean>(() => {
-    return props.invocation?.messages.length ? true : false;
+const uniqueMessages = computed(() => {
+    const messages = props.invocation?.messages || [];
+    const uniqueMessagesSet = new Set(messages.map((message) => JSON.stringify(message)));
+    return Array.from(uniqueMessagesSet).map((message) => JSON.parse(message)) as InvocationMessageResponseModel[];
```

Comment:
```suggestion
    const uniqueMessagesSet = new Set(messages.map((message) => JSON.stringify(message)));
    return Array.from(uniqueMessagesSet).map((message) => JSON.parse(message)) as typeof messages;
```

That would avoid any drift, assuming that stringify/parse doesn't change the type.

Reply:
![image](https://github.com/user-attachments/assets/00e92fb0-e9bb-418e-a6f9-72f8d024538c)

is the reason I assigned the `InvocationMessageResponseModel` type. And no, it is not stringify/parse changing the type, the error is being raised outside of this branch as well if you go to `WorkflowInvocationOverview`

Reply:
Have you tried my suggestion ? If this fails (it compiles fine for me) then I think there's something else wrong.

Reply:
That's what I tried in the screenshot. There might be something off on my end then 🤔 , I'll push your suggestion and you could confirm it compiles fine?

Reply:
It compiles fine, otherwise the tests would fail. I do see this
<img width="1503" alt="Screenshot 2024-08-27 at 17 07 32" src="https://github.com/user-attachments/assets/e28e2569-7919-4532-bc18-fc219c3d60cd">

however. The solution to a problem like this can't be re-assigning the type ... this will never fail any compile-time checks, which we really have to avoid.

I think the proper fix is to drop the (semi-)hand-generated invocation interfaces in client/src/components/WorkflowInvocationState/invocationMessageModel.ts. These were only needed when we didn't have a fully-typed invocation model in the openapi docs, which we do now. I don't think there's actually a type-mismatch and it's a vetur issue, but of course it's cleaner to drop this.

I'll push a fix for that.

Reply:
That makes sense, thank you!

Reply:
Turns out we already had some type drift: https://github.com/galaxyproject/galaxy/pull/18730/commits/e9eba0c620a3fb320b53b37a50403d31fa877763